core/tests/puppeteer-acceptance-tests/user-utilities/super-admin-utils.js

```diff
@@ -101,4 +101,444 @@ module.exports = class e2eSuperAdmin extends baseUser {
     showMessage(`User ${username} does not have the ${role} role!`);
     await this.goto(currentPageUrl);
   }
+
+  async editClassroom({ topicId }) {
+    await this.goto('http://localhost:8181/admin#/config');
+
+    await this.clickOn('.e2e-test-add-list-entry');
+    // TODO(#19668): Remove the for loop that types in the entire topic id.
+    for (let i = 0; i < topicId.length; i++) {
+      await this.type(
+        '.e2e-test-schema-based-dict-editor ' +
+        '.e2e-test-schema-based-list-editor-table-data ' +
+        'input[type="text"]',
+        topicId[i], { visible: true });
+    }
+
+    await this.page.on('dialog', async(dialog) => {
+      await dialog.accept();
+    });
+    await this.clickOn('.e2e-test-save-all-configs');
+    await this.page.waitForSelector(
+      '.e2e-test-status-message', { visible: true });
+  }
+
+  async createTopic({
+    name, urlFragment, webTitleFragment, description, thumbnail, metaContent,
+    assignedSkills, subtopics, diagnosticTestSkills, isPublished
+  }) {
+    await this.goto('http://localhost:8181/topics-and-skills-dashboard');
+
+    await this.page.waitForSelector(
+      '.e2e-test-no-topics-and-no-skills-present-message', { hidden: true });
+    await this.page.waitForSelector(
+      '.e2e-test-topics-and-skills-tab-list', { visible: true });
+    await this.clickOn('.e2e-test-topics-tab');
+
+    await this.clickOn('.e2e-test-create-topic-button', { visible: true });
+    await this.withinModal({
+      selector: 'oppia-create-new-topic-modal',
+      whenOpened: async(_this) => {
+        await _this.type('.e2e-test-new-topic-name-field', name);
+        await _this.type(
+          '.e2e-test-new-topic-url-fragment-field', urlFragment);
+        await _this.type(
+          '.e2e-test-new-page-title-fragm-field', webTitleFragment);
+
+        await _this.clickOn('.e2e-test-photo-button');
+        await _this.withinModal({
```

Comment:
Functions in this codebase start with verbs; maybe we can have this be doWithinModal or similar?

Reply:
The feature related to this review thread was removed.